### ISSUE TITLE
No image is loaded when using the GPU version.

### ISSUE THREAD

**Issue by hubutui** [jsbroks/coco-annotator#504]
step to reproduce:
1. modify, add these lines to `backend/Dockerfile`:
  ```text
RUN apt install -y \
    libgl1 \
    libglib2.0-0 \
    libsm6 \
    libxext6 \
    libxrender-dev
  ```
to fix `import cv2` error.
2. run `bash build_gpu.sh`
3. `docker-compose -f docker-compose.gpu.yml up --build -d`

However, after creating a dataset and add images to the dataset. I could not see any image shown.
I can confirm that `docker-compose up --build -d` works.

**Comment by hubutui** [comment]
I attach to the mongodb container, and run:
```bash
mongo
use flask
db.image_model.find
```
and nothing is printed. It seems the images are not added to the mongodb?

**Comment by hubutui** [comment]
If a add some images to the dataset by copy them to `datasets/cars`, then only one image is added:
```text
annotator_webserver.gpu | [2021-09-03 04:39:00 +0000] [12] [INFO] [File Watcher]: File /datasets/cars/cars.bak/00f88c4f0a/00000.jpg for created
annotator_webserver.gpu | [2021-09-03 04:39:00 +0000] [12] [INFO] [File Watcher]: Adding new file to database: /datasets/cars/cars.bak/00f88c4f0a/00000.jpg
```
and `db.image_model.find()` ouputs:
```text
{ "_id" : 1, "dataset_id" : 1, "category_ids" : [ ], "path" : "/datasets/cars/cars.bak/00f88c4f0a/00000.jpg", "width" : 1280, "height" : 720, "file_name" : "00000.jpg", "annotated" : false, "annotating" : [ ], "num_annotations" : 0, "metadata" : {  }, "deleted" : false, "milliseconds" : 0, "events" : [ ], "regenerate_thumbnail" : true }
```

**Comment by hubutui** [comment]
Here are the `pip list` result for cpu and gpu python-env.
<details>
  <summary>cpu version, python 3.6.8</summary>
<pre>
Package                       Version           
----------------------------- ------------------
absl-py                       0.7.1             
alabaster                     0.7.12            
amqp                          2.4.2             
aniso8601                     6.0.0             
antlr4-python3-runtime        4.8               
appdirs                       1.4.4             
argh                          0.26.2            
astor                         0.7.1             
atomicwrites                  1.3.0             
attrs                         19.1.0            
Babel                         2.6.0             
backcall                      0.1.0             
billiard                      3.5.0.5           
black                         21.4b2            
bleach                        3.1.0             
celery                        4.2.2             
certifi                       2019.3.9          
chardet                       3.0.4             
click                         8.0.1             
cloudpickle                   1.6.0             
cycler                        0.10.0            
Cython                        0.29.6            
dataclasses                   0.8               
decorator                     4.4.0             
defusedxml                    0.5.0             
detectron2                    0.5+cu102         
dextr                         0.0.1             
dnspython                     1.16.0            
docutils                      0.14              
entrypoints                   0.3               
eventlet                      0.24.1            
Flask                         1.0.2             
Flask-Cors                    3.0.7             
Flask-Login                   0.4.1             
flask-mongoengine             0.9.5             
flask-restplus                0.12.1            
Flask-SocketIO                3.3.2             
Flask-WTF                     0.14.2            
future                        0.18.2            
fvcore                        0.1.5.post20210825
gast                          0.2.2             
google-images-download        2.5.0             
greenlet                      0.4.15            
grpcio                        1.19.0            
gunicorn                      19.9.0            
h5py                          2.9.0             
hydra-core                    1.1.1             
idna                          2.8               
imageio                       2.5.0             
imagesize                     1.1.0             
imantics                      0.1.9             
imgaug                        0.2.8             
importlib-metadata            4.8.1             
importlib-resources           5.2.2             
iopath                        0.1.8             
ipykernel                     5.1.0             
ipyparallel                   6.2.3             
ipython                       7.4.0             
ipython-genutils              0.2.0             
ipywidgets                    7.4.2             
itsdangerous                  1.1.0             
jedi                          0.13.3            
Jinja2                        2.10              
jsonschema                    3.0.1             
jupyter                       1.0.0             
jupyter-client                5.2.4             
jupyter-console               6.0.0             
jupyter-core                  4.4.0             
Keras                         2.2.4             
Keras-Applications            1.0.7             
Keras-Preprocessing           1.0.9             
kiwisolver                    1.0.1             
kombu                         4.3.0             
lxml                          4.3.3             
Markdown                      3.1               
MarkupSafe                    1.1.1             
mask-rcnn                     2.1               
matplotlib                    3.0.3             
mistune                       0.8.4             
mock                          2.0.0             
mongoengine                   0.17.0            
monotonic                     1.5               
more-itertools                7.0.0             
mypy-extensions               0.4.3             
nbconvert                     5.4.1             
nbformat                      4.4.0             
networkx                      2.2               
nose                          1.3.7             
notebook                      5.7.8             
numpy                         1.16.2            
omegaconf                     2.1.1             
opencv-python                 4.0.0.21          
packaging                     19.0              
pandocfilters                 1.4.2             
parso                         0.3.4             
pathspec                      0.9.0             
pathtools                     0.1.2             
pbr                           5.1.3             
pexpect                       4.6.0             
pickleshare                   0.7.5             
Pillow                        8.3.1             
pip                           19.0.3            
pluggy                        0.9.0             
portalocker                   2.3.2             
prometheus-client             0.6.0             
prompt-toolkit                2.0.9             
protobuf                      3.7.1             
ptyprocess                    0.6.0             
py                            1.8.0             
pycocotools                   2.0.2             
pydot                         1.4.2             
Pygments                      2.3.1             
pymongo                       3.7.2             
pyparsing                     2.3.1             
pyrsistent                    0.14.11           
pytest                        3.9.3             
pytest-ordering               0.6               
python-dateutil               2.8.0             
python-engineio               3.5.0             
python-socketio               3.1.2             
pytz                          2018.9            
PyWavelets                    1.0.2             
PyYAML                        5.1               
pyzmq                         18.0.1            
qtconsole                     4.4.3             
regex                         2021.8.28         
requests                      2.21.0            
scikit-image                  0.15.0            
scikit-learn                  0.20.3            
scipy                         1.2.1             
selenium                      3.141.0           
Send2Trash                    1.5.0             
setuptools                    40.8.0            
Shapely                       1.6.4.post2       
six                           1.12.0            
snowballstemmer               1.2.1             
Sphinx                        2.0.0             
sphinxcontrib-applehelp       1.0.1             
sphinxcontrib-devhelp         1.0.1             
sphinxcontrib-htmlhelp        1.0.1             
sphinxcontrib-jsmath          1.0.1             
sphinxcontrib-qthelp          1.0.2             
sphinxcontrib-serializinghtml 1.1.3             
tabulate                      0.8.9             
tensorboard                   1.13.1            
tensorflow                    1.13.1            
tensorflow-estimator          1.13.0            
termcolor                     1.1.0             
terminado                     0.8.2             
testpath                      0.4.2             
toml                          0.10.2            
torch                         1.9.0             
torchvision                   0.10.0            
tornado                       6.0.2             
tqdm                          4.62.2            
traitlets                     4.3.2             
typed-ast                     1.4.3             
typing-extensions             3.10.0.1          
urllib3                       1.24.1            
vine                          1.3.0             
watchdog                      0.8.3             
wcwidth                       0.1.7             
webencodings                  0.5.1             
Werkzeug                      0.15.2            
wheel                         0.33.1            
widgetsnbextension            3.4.2             
WTForms                       2.2.1             
yacs                          0.1.8             
zipp                          3.5.0             
</pre>
</details>

<details>
  <summary>gpu version, python 3.6.8</summary>
  <pre>
Package                       Version           
----------------------------- ------------------
absl-py                       0.7.1             
alabaster                     0.7.12            
amqp                          2.6.1             
aniso8601                     9.0.1             
antlr4-python3-runtime        4.8               
appdirs                       1.4.4             
argh                          0.26.2            
asn1crypto                    0.24.0            
astor                         0.8.0             
attrs                         19.1.0            
Babel                         2.9.1             
backcall                      0.1.0             
billiard                      3.5.0.5           
black                         21.4b2            
bleach                        3.1.0             
celery                        4.2.2             
certifi                       2021.5.30         
charset-normalizer            2.0.4             
click                         8.0.1             
cloudpickle                   1.6.0             
cryptography                  2.1.4             
cycler                        0.10.0            
Cython                        0.29.6            
dataclasses                   0.8               
decorator                     4.4.0             
defusedxml                    0.6.0             
detectron2                    0.5+cu102         
dextr                         0.0.1             
dnspython                     2.1.0             
docutils                      0.17.1            
entrypoints                   0.3               
enum34                        1.1.6             
eventlet                      0.25.1            
Flask                         1.0.2             
Flask-Cors                    3.0.7             
Flask-Login                   0.4.1             
flask-mongoengine             0.9.5             
flask-restplus                0.12.1            
Flask-SocketIO                3.3.2             
Flask-WTF                     0.14.2            
future                        0.18.2            
fvcore                        0.1.5.post20210825
gast                          0.2.2             
google-images-download        2.5.0             
google-pasta                  0.1.7             
greenlet                      0.4.15            
grpcio                        1.21.1            
gunicorn                      19.9.0            
h5py                          2.9.0             
hydra-core                    1.1.1             
idna                          2.6               
imageio                       2.5.0             
imagesize                     1.1.0             
imantics                      0.1.9             
imgaug                        0.2.8             
importlib-metadata            4.8.1             
importlib-resources           5.2.2             
iniconfig                     1.1.1             
iopath                        0.1.8             
ipykernel                     5.1.1             
ipython                       7.5.0             
ipython-genutils              0.2.0             
ipywidgets                    7.4.2             
itsdangerous                  2.0.1             
jedi                          0.13.3            
Jinja2                        2.10.1            
joblib                        1.0.1             
jsonschema                    3.0.1             
jupyter                       1.0.0             
jupyter-client                5.2.4             
jupyter-console               6.0.0             
jupyter-core                  4.5.0             
jupyter-http-over-ws          0.0.6             
Keras                         2.2.4             
Keras-Applications            1.0.8             
Keras-Preprocessing           1.1.0             
keyring                       10.6.0            
keyrings.alt                  3.0               
kiwisolver                    1.1.0             
kombu                         4.3.0             
lxml                          4.6.3             
Markdown                      3.1.1             
MarkupSafe                    1.1.1             
mask-rcnn                     2.1               
matplotlib                    3.1.0             
mistune                       0.8.4             
mongoengine                   0.17.0            
monotonic                     1.6               
mypy-extensions               0.4.3             
nbconvert                     5.5.0             
nbformat                      4.4.0             
networkx                      2.5.1             
nose                          1.3.7             
notebook                      5.7.8             
numpy                         1.16.4            
omegaconf                     2.1.1             
opencv-python                 4.0.0.21          
packaging                     21.0              
pandocfilters                 1.4.2             
parso                         0.4.0             
pathspec                      0.9.0             
pathtools                     0.1.2             
pexpect                       4.7.0             
pickleshare                   0.7.5             
Pillow                        8.3.1             
pip                           19.1.1            
pluggy                        1.0.0             
portalocker                   2.3.2             
prometheus-client             0.7.1             
prompt-toolkit                2.0.9             
protobuf                      3.8.0             
ptyprocess                    0.6.0             
py                            1.10.0            
pycocotools                   2.0.2             
pycrypto                      2.6.1             
pydot                         1.4.2             
Pygments                      2.4.2             
pygobject                     3.26.1            
pymongo                       3.7.2             
pyparsing                     2.4.0             
pyrsistent                    0.15.2            
pytest                        6.2.5             
pytest-ordering               0.6               
python-apt                    1.6.4             
python-dateutil               2.8.0             
python-engineio               4.2.1             
python-socketio               3.1.2             
pytz                          2021.1            
PyWavelets                    1.1.1             
pyxdg                         0.25              
PyYAML                        5.4.1             
pyzmq                         18.0.1            
qtconsole                     4.5.1             
regex                         2021.8.28         
requests                      2.26.0            
scikit-image                  0.17.2            
scikit-learn                  0.24.2            
scipy                         1.5.4             
SecretStorage                 2.3.1             
selenium                      3.141.0           
Send2Trash                    1.5.0             
setuptools                    41.0.1            
Shapely                       1.7.1             
six                           1.11.0            
snowballstemmer               2.1.0             
Sphinx                        1.8.5             
sphinxcontrib-serializinghtml 1.1.5             
sphinxcontrib-websupport      1.2.4             
tabulate                      0.8.9             
tensorboard                   1.14.0            
tensorflow                    1.14.0            
tensorflow-estimator          1.14.0            
tensorflow-gpu                1.14.0            
termcolor                     1.1.0             
terminado                     0.8.2             
testpath                      0.4.2             
threadpoolctl                 2.2.0             
tifffile                      2020.9.3          
toml                          0.10.2            
toolz                         0.9.0             
torch                         1.9.0             
torchvision                   0.10.0            
tornado                       6.0.2             
tqdm                          4.62.2            
traitlets                     4.3.2             
typed-ast                     1.4.3             
typing-extensions             3.10.0.2          
urllib3                       1.24.1            
vine                          1.3.0             
watchdog                      0.8.3             
wcwidth                       0.1.7             
webencodings                  0.5.1             
Werkzeug                      0.15.4            
wheel                         0.30.0            
widgetsnbextension            3.4.2             
wrapt                         1.11.2            
WTForms                       2.2.1             
yacs                          0.1.8             
zipp                          3.5.0             
<pre>
</details>
